pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -20,14 +21,25 @@ import (
 	"github.com/rancher/types/config"
 	"github.com/rancher/types/user"
 	"github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
+	k8stypes "k8s.io/apimachinery/pkg/types"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
 const AgentForceDeployAnn = "io.cattle.agent.force.deploy"
 
+var (
+	nodeAgentImages    map[k8stypes.UID]string
+	nodeMutex          sync.RWMutex
+	clusterAgentImages map[k8stypes.UID]string
+	clusterMutex       sync.RWMutex
+)
+
 func Register(ctx context.Context, management *config.ManagementContext, clusterManager *clustermanager.Manager) {
+	nodeAgentImages = make(map[k8stypes.UID]string)
+	clusterAgentImages = make(map[k8stypes.UID]string)
```

Comment:
Instantiate the vars in the declaration, if someone calls register twice this will overwrite